mlir/lib/CAPI/Dialect/Rock.cpp

```diff
@@ -0,0 +1,79 @@
+//===- MIGraphX.cpp - C Interface for MIGraphX dialect---------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir-c/Dialect/Rock.h"
+#include "mlir/CAPI/IR.h"
+#include "mlir/CAPI/Pass.h"
+#include "mlir/CAPI/Registration.h"
+#include "mlir/CAPI/Wrap.h"
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/Tuning/ConvContext.h"
+#include "mlir/Dialect/Rock/Tuning/RockTuning.h"
+
+#include <vector>
+
+MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(Rock, rock, mlir::rock::RockDialect)
+DEFINE_C_API_PTR_METHODS(MlirRockTuningSpace, mlir::rock::TunableParams)
+DEFINE_C_API_PTR_METHODS(MlirRockTuningParam, mlir::rock::ParamEntry)
+// DEFINE_C_API_METHOD(MlirRockTuningKey, const void);
+
+using namespace mlir;
+
+MLIR_CAPI_EXPORTED MlirRockTuningSpace
+mlirRockTuningSpaceCreate(MlirModule module) {
+  struct rock::TunableParams *newParams;
+  auto mod = unwrap(module);
+  newParams = rock::createTunableParams(mod);
+  return wrap(newParams);
+}
+
+MLIR_CAPI_EXPORTED int
+mlirRockTuningGetNumParamsQuick(MlirRockTuningSpace params) {
+  auto tuningSpace = unwrap(params);
+  return tuningSpace->numHeuristicQuick;
+}
+
+MLIR_CAPI_EXPORTED
+int mlirRockTuningGetNumParamsFull(MlirRockTuningSpace params) {
+  auto tuningSpace = unwrap(params);
+  return tuningSpace->tuningRange.size();
+}
+
+// Allocate parameter entry to hold string allocation and pass it more easily.
+MLIR_CAPI_EXPORTED MlirRockTuningParam
+mlirRockTuningCreateParamAt(MlirRockTuningSpace params, int pos) {
+  auto tuningSpace = unwrap(params);
+  rock::ParamEntry *param = new rock::ParamEntry();
+  param->param = tuningSpace->tuningRange[pos];
```

Comment:
This is not defensive enough. A client can pass in a `pos` larger than the size of range therefore crashing our plugin.

Reply:
That's good point, we should cover that.